lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2966,6 +2970,33 @@ void IRGenDebugInfoImpl::emitImport(ImportDecl *D) {
   ImportedModules.insert(Imported.importedModule);
 }
 
+/// This is effectively \p clang::CGDebugInfo::getCallSiteRelatedAttrs().
+llvm::DINode::DIFlags IRGenDebugInfoImpl::getCallSiteRelatedAttrs() const {
+  auto Optimize = IGM.getClangASTContext().getLangOpts().Optimize;
+  auto Loader = IGM.getSILModule().getASTContext().getClangModuleLoader();
+  auto *Importer = static_cast<ClangImporter *>(&*Loader);
+  auto &CGO = Importer->getCodeGenOpts();
+
+  // Call site-related attributes are only useful in optimized programs, and
+  // when there's a possibility of debugging backtraces.
```

Comment:
This sentence doesn't make sense...

Reply:
Regardless of the wording. @JDevlieghere  Wouldn't the linker deduplicate symbols even at -Onone? Is there a good argument to do this only in optimized code?

Reply:
clang only does it in optimized code

Reply:
Yes, the linker doesn't care (or even know) about the optimization level. I assume the equivalent code in clang is thinking about function merging in the compiler?

Reply:
This is all the clang side code does, there is nothing about function merging mentioned

```
  // Call site-related attributes are only useful in optimized programs, and
  // when there's a possibility of debugging backtraces.
  if (!CGM.getLangOpts().Optimize || DebugKind == codegenoptions::NoDebugInfo ||
      DebugKind == codegenoptions::LocTrackingOnly)
    return llvm::DINode::FlagZero;
```

But I can just remove this check, in swift.

Reply:
I think we should consider changing this behavior on Darwin in Clang, too, since it's a linker-specific thing on Darwin only.